docs/cookbook/images/image-on-entity.rst

```diff
@@ -0,0 +1,376 @@
+     * @ORM\Table(name="sylius_payment_method")
+     */
+    #[ORM\Entity]
+    #[ORM\Table(name: 'sylius_payment_method')]
+    class PaymentMethod extends BasePaymentMethod implements ImageAwareInterface
+    {
+        /**
+         * @Assert\Valid
+         * @ORM\OneToOne(mappedBy="owner", targetEntity="App\Entity\Payment\PaymentMethodImage", cascade={"all"}, orphanRemoval=true)
+         */
+        #[ORM\OneToOne(mappedBy: 'owner', targetEntity: PaymentMethodImage::class, cascade: ['all'], orphanRemoval: true)]
+        protected ?PaymentMethodImage $image = null;
+
+        /** @return PaymentMethodImage|null */
+        public function getImage(): ?ImageInterface
+        {
+            return $this->image;
+        }
+
+        /** @var PaymentMethodImage|null $image */
+        public function setImage(?ImageInterface $image): void
+        {
+            $image?->setOwner($this);
+
+            $this->image = $image;
+        }
+    }
+
+3. Create the image form type
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The constructor arguments are inlined to facilitate the autowiring, but you may inject them from the service configuration if necessary, an example of that is provided in the next step.
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Form\Type;
+
+    use App\Entity\Payment\PaymentMethodImage;
+    use Sylius\Bundle\CoreBundle\Form\Type\ImageType;
+    use Symfony\Component\Form\FormBuilderInterface;
+
+    final class PaymentMethodImageType extends ImageType
+    {
+        public function __construct()
+        {
+            parent::__construct(PaymentMethodImage::class, ['sylius']);
+        }
+
+        public function buildForm(FormBuilderInterface $builder, array $options): void
+        {
+            parent::buildForm($builder, $options);
+
+            $builder->remove('type');
+        }
+
+        public function getBlockPrefix(): string
+        {
+            return 'payment_method_image';
+        }
+    }
+
+Register the image form type service if necessary, for example in case the autowiring or autoconfiguration are disabled or you wish to inject the constructor arguments from configuration:
+
+.. code-block:: yaml
+
+    # services.yml or another configuration file of your choice
+    services:
+        App\Form\Type\PaymentMethodImageType:
+    # Removing the __constructor from the PaymentMethodImageType and configure the arguments if necessary
+    #        arguments:
+    #            - '%app.model.payment_method_image.class%'
+    #            - ['sylius']
+            tags:
+                - { name: form.type }
+
+4. Configure the image resource
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+
+    # config/packages/_sylius.yaml
+    sylius_resource:
+        resources:
+            app.payment_method_image:
+                classes:
+                    model: App\Entity\Payment\PaymentMethodImage
+                    form: App\Form\Type\PaymentMethodImageType
```

Comment:
Don't you also need to update the `PaymentMethod` resource config?

Reply:
No, the Payment Method resource configuration stays the same.